lambdas/data-migration1/tests/test-rules.js

```diff
@@ -0,0 +1,390 @@
+const { getKnexClient, localStackConnectionEnv } = require('@cumulus/db');
+const { randomId, randomString } = require('@cumulus/common/test-utils');
+
+// eslint-disable-next-line node/no-unpublished-require
+const { migrationDir } = require('../../db-migration');
+const { migrateRuleRecord, migrateRules, getCumulusId } = require('../dist/lambda/rules');
+const { RecordAlreadyMigrated, ColumnDoesNotExist } = require('../dist/lambda/errors');
+
+const testDbName = `data_migration_1_${cryptoRandomString({ length: 10 })}`;
+const testDbUser = 'postgres';
+const workflow = randomId('workflow-');
+const ruleOmitList = ['createdAt', 'updatedAt', 'state', 'provider', 'collection', 'rule'];
+
+const generateFakeRule = (collectionName, collectionVersion, providerId, enabled = true) => ({
+  name: cryptoRandomString({ length: 10 }),
+  workflow: workflow,
+  provider: providerId,
+  state: enabled ? 'ENABLED' : 'DISABLED',
+  collection: {
+    name: collectionName,
+    version: collectionVersion,
+  },
+  rule: { type: 'onetime', value: randomString(), arn: randomString(), logEventArn: randomString() },
+  executionNamePrefix: randomString(),
+  meta: { key: 'value' },
+  queueUrl: randomString(),
+  payload: { result: { key: 'value' } },
+  tags: ['tag1', 'tag2'],
+  createdAt: Date.now(),
+  updatedAt: Date.now(),
+});
+
+const migrateFakeCollectionRecord = async (record, knex) => {
+  const updatedRecord = translateApiCollectionToPostgresCollection(record);
+  await knex('collections').insert(updatedRecord);
+};
+
+const fakeEncryptFunction = async () => 'fakeEncryptedString';
+
+const migrateFakeProviderRecord = async (record, knex) => {
+  const updatedRecord = await translateApiProviderToPostgresProvider(record, fakeEncryptFunction);
+  await knex('providers').insert(updatedRecord);
+};
+
+let collectionsModel;
+let providersModel;
+let rulesModel;
+let fakeCollection;
+let fakeProvider;
+
+test.before(async (t) => {
+  process.env.stackName = cryptoRandomString({ length: 10 });
+  process.env.system_bucket = cryptoRandomString({ length: 10 });
+  process.env.CollectionsTable = cryptoRandomString({ length: 10 });
+  process.env.ProvidersTable = cryptoRandomString({ length: 10 });
+  process.env.RulesTable = cryptoRandomString({ length: 10 });
+
+  const workflowfile = `${process.env.stackName}/workflows/${workflow}.json`;
+  const messageTemplateKey = `${process.env.stackName}/workflow_template.json`;
+
+  collectionsModel = new Collection();
+  await collectionsModel.createTable();
+
+  providersModel = new Provider();
+  await providersModel.createTable();
+
+  rulesModel = new Rule();
+  await rulesModel.createTable();
+  await createBucket(process.env.system_bucket);
+
+  fakeCollection = fakeCollectionFactory();
+  fakeProvider = fakeProviderFactory({
+    encrypted: true,
+    privateKey: 'key',
+    cmKeyId: 'key-id',
+    certificateUri: 'uri',
+    createdAt: new Date(2020, 11, 17),
+  });
```

Comment:
It seems like you're doing this in several of the tests, so I would move these calls to a `beforeEach` hook and attach them to `t.context` (e.g. `t.context.fakeCollection`). Then, in any given test, you can access `t.context.fakeCollection` rather than having to create it in the test setup.